streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/Materialized.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.scala
+
+import org.apache.kafka.common.serialization.Serde
+import org.apache.kafka.common.utils.Bytes
+import org.apache.kafka.streams.kstream.{Materialized => JMaterialized}
+import org.apache.kafka.streams.processor.StateStore
+import org.apache.kafka.streams.state._
+
+object Materialized {
```

Comment:
I would go defensive with implicits on `Materialized`. First, this abstraction doesn't add much value since it returns the Java abstraction. Also if someone doesn't wish to pass a serde to a `Materialized`, and use `Materialized.as` they can be inadvertently passed if implicits are in scope. This will lead to errors that are difficult to debug.